### PR TITLE
#3136 Add hasInitialConditions_ member to DataInterface

### PR DESCRIPTION
**Pull request by gaouizer**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h [hunk 1]
```diff
@@ -80,6 +81,12 @@ class ComponentInterface {
    */
   void hasDynamicModel(bool hasDynamicModel);
 
+  /**
+   * @brief Setter for the component's hasInitialConditions_ attribute
+   * @param hasInitialConditions @b true is component has initial conditions set, @b false else
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h:86]: true if

File: dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h [hunk 2]
```diff
@@ -170,6 +177,12 @@ class ComponentInterface {
    */
   virtual int getComponentVarIndex(const std::string& varName) const = 0;
 
+  /**
+   * @brief Getter for the component's hasInitialConditions_ attribute
+   * @return  @b true is component has initial conditions set, @b false else
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h:182]: true if

File: dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h [hunk 3]
```diff
@@ -218,6 +231,7 @@ class ComponentInterface {
  private:
   bool hasDynamicModel_;  ///< @b true is component has a dynamic model (other than c++ one), @b false else
   boost::shared_ptr<SubModel> modelDyn_;  ///< dynamic model of the component
+  bool hasInitialConditions_;  ///< @b true is component has initial conditions set, @b false else
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h:234]: true if

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestGeneratorInterface.cpp [hunk 4]
```diff
@@ -80,6 +80,10 @@ TEST(DataInterfaceTest, Generator_1) {
   genItf.setVoltageLevelInterface(vlItf);
   ASSERT_EQ(genItf.getID(), "GEN1");
 
+  ASSERT_FALSE(genItf.hasInitialConditions());
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestGeneratorInterface.cpp:83]: I'd try as much as possible to test the hasInitialConditions setting P,Q,V,angle (it depends on the component) via myComponent.getTerminal().setBlaBla

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Review by marcochiaramello — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Comment by rosiereflo** [comment]
Could you please squash the commits?

**Pull request merged**